### PR TITLE
Initial changes to backend.

### PR DESCRIPTION
**Pull request by pmeskridge**
# To be Completed by the Pull Requester

### Checklist:

- [x] I have performed a self-review of my code to ensure that unintended code is not introduced.
- [x] I have commented my code, particularly in hard-to-understand areas to ensure that the next group of developers can understand my code.
- [X] I have added tests (if appropriate) that prove my fix/feature functions so that future modifications to the code do not break my additions.

### Description:

- Added Status enumerated field to hold the status of user accounts as: Requested, Enabled, Disabled or Denied.  Altered code need to login/verify user account to use the Status field in place/or addition to the Enabled field implemented in Waypoint.  Also added a Status to the seed and repo accounts created.

### Type of change (Pick Any)

2. New feature. Non-breaking change which adds functionality, example: new page on the frontend.

### How Has This Been Tested?

Test added to e2e test to use a requested account to login with and should return a wrong email/password error.

---

# To be Completed By Reviewers

Reviewer # 01

- [x] I have performed a review of the code and checked for possible unintentional edits.
- [x] I concur with the amount of comments made.
- [x] I concur with the amount of tests made.

Reviewer # 02

- [x] I have performed a review of the code and checked for possible unintentional edits.
- [x] I concur with the amount of comments made.
- [x] I concur with the amount of tests made.

### CODE REVIEW

File: apps/ods-api/prisma-seed.ts [hunk 1]
```diff
@@ -311,7 +312,7 @@ async function main() {
         firstName: 'Henry',
         lastName: 'Henderson',
         grade: 'O-5',
-        enabled: true,
+        status: 'ENABLED',
```
Comment (yurisim) [inline apps/ods-api/prisma-seed.ts:315]: Just note there's a difference in Henry Henderson's status in the seed prod versus the seed here. 
```status: 'ENABLED'```

versus

```status: 'REQUESTED'```

File: apps/ods/src/app/request-account/request-account.component.ts [hunk 2]
```diff
@@ -65,6 +66,7 @@ export class RequestAccountComponent implements OnInit {
   ];
   orgs: Observable<string[]>;
   roles = ['Commander', 'DEI', 'Admin'];
+  status = Status; //initial status when an account is requested
```
Comment (yurisim) [inline apps/ods/src/app/request-account/request-account.component.ts:69]: I'm dumb, where is this Status initially defined on this document?

Comment (yurisim) [reply apps/ods/src/app/request-account/request-account.component.ts:69]: Capital Status

File: libs/auth/src/auth/auth.service.ts [hunk 3]
```diff
@@ -60,12 +60,13 @@ export class AuthService {
...
   }
 
+  //Added user.status to accomodate inclusion of field in ODS as the least intrusive way
   async validateUser(
     emailOrUsername: string,
     passwordPlaintextInput: string
   ): Promise<unknown> {
     const user = await this.getUserByEmailOrUsername(emailOrUsername);
-    if (user && user.enabled) {
+    if (user && (user.enabled || user.status == 'ENABLED')) {
```
Comment (yurisim) [inline libs/auth/src/auth/auth.service.ts:69]: Did you get a chance to clear this up with Cater? 

You can resolve this comment either way. But if you didn't please do so after he gets back. I'm okay w/ merging this into dev tho.

File: libs/auth/src/auth/interfaces/user-service.interface.ts [hunk 4]
```diff
@@ -4,12 +4,14 @@ import {
 } from './refreshToken.interface';
 
 // TODO: May need to refactor once we get generator going
-
+//Added status to accomodate inclusion in ODS and made enabled optional
```
Comment (yurisim) [inline libs/auth/src/auth/interfaces/user-service.interface.ts:7]: Clear this w/ Cater as well when he gets back

File: libs/auth/src/auth/strategies/jwt.strategy.ts [hunk 5]
```diff
@@ -24,8 +24,9 @@ export class JWTStrategy extends PassportStrategy(Strategy, 'jwt') {
     // but again, this centralizes it
 
     //TODO [ODST-305] client fingerprint?
+    //added user.status to accomodate status field in ODS
     const user = await this.userService.findUnique({ id: payload.sub });
-    if (user && user.enabled) {
+    if (user && (user.enabled || user.status == 'ENABLED')) {
```
Comment (yurisim) [inline libs/auth/src/auth/strategies/jwt.strategy.ts:29]: same as above

**Review by yurisim — APPROVED** [review]

File: apps/ods-api/src/user/user.resolver.ts [hunk 6]
```diff
@@ -43,6 +43,7 @@ export class UserResolver {
           grade: true,
           firstName: true,
           lastName: true,
+          status: true,
```
Comment (BrendenMcGrew) [inline apps/ods-api/src/user/user.resolver.ts:46]: Unused variable earlier in file

**Review by BrendenMcGrew — APPROVED** [review]

File: apps/ods-api/prisma-seed.ts (hunk 1 above)
Comment (pmeskridge) [reply apps/ods-api/prisma-seed.ts:315]: Status should be REQUESTED for both.

File: apps/ods/src/app/request-account/request-account.component.ts (hunk 2 above)
Comment (pmeskridge) [reply apps/ods/src/app/request-account/request-account.component.ts:69]: "Status" is an enum defined in the types.graphql and imported on line 5.  In the "submit" function, on line 119 "status" is set to the enum value "REQUESTED".

File: libs/auth/src/auth/auth.service.ts (hunk 3 above)
Comment (pmeskridge) [reply libs/auth/src/auth/auth.service.ts:69]: Did not resolve before he left as he seemed preoccupied.  Will do so at the first available opportunity.

File: libs/auth/src/auth/interfaces/user-service.interface.ts (hunk 4 above)
Comment (pmeskridge) [reply libs/auth/src/auth/interfaces/user-service.interface.ts:7]: Okay.

File: libs/auth/src/auth/strategies/jwt.strategy.ts (hunk 5 above)
Comment (pmeskridge) [reply libs/auth/src/auth/strategies/jwt.strategy.ts:29]: Okay.

**Pull request merged**